website/docs/guides/ip_spaces.html.markdown

```diff
@@ -0,0 +1,286 @@
+  the service provider* through a quota-based system. 
+* **Shared** IP Space - An IP Space for services and management networks that are required in the
+  tenant space, but as a service provider, you don't want to expose it to organizations in your
+  environment. Simply put, *only provider* can *perform IP allocations* from such IP Space.
+* **Private** IP Space - Private IP Spaces are dedicated to a single tenant - a private IP space is
+  used by only one organization that is specified during the space creation. For this organization,
+  IP consumption is unlimited.
+
+
+## List of resources with IP Space support
+
+The following resources for IP Space management are available with Terraform Provider for
+VCD starting with version 3.10:
+
+* `vcd_ip_space` - provides IP Space and default quota definition capability
+* `vcd_ip_space_uplink` - provides capability to assign IP Space Uplink for Provider Gateways
+  (resource `vcd_external_network_v2`)
+* `vcd_ip_space_ip_allocation` - provides capability to allocate floating IPs or IP Prefixes
+* `vcd_ip_space_custom_quota` - provides capability to set Org specific Custom Quotas and override
+  default ones defined in `vcd_ip_space`
+* `vcd_external_network_v2` - fields `use_ip_spaces` and `dedicated_org_id` (applicable only to
+  T0 or T0 VRF backed networks also known as Provider Gateways in UI)
+* `vcd_nsxt_edgegateway` - none of the fields `subnet_with_total_ip_count`, `subnet`,
+  `subnet_with_ip_count` are mandatory when specifying `external_network_id` that is using IP
+  Spaces. As a result they will not be populated after read operations together with `used_ip_count`
+  and `unused_ip_count`. Additional computed flag `use_ip_spaces` to tell if the Edge Gateway is
+  using IP Spaces (is backed by Provider Gateway that has IP Space Uplinks)
+
+
+## Sample end to end configuration using IP Spaces
+
+-> There are new rights for IP Space management starting with VCD 10.4.1. Some of them are [listed
+in the
+prerequisites](https://docs.vmware.com/en/VMware-Cloud-Director/10.4/VMware-Cloud-Director-Service-Provider-Admin-Portal-Guide/GUID-575513A8-9ADE-4A3D-92AB-CB0917FF8316.html)
+for IP Space management in Organizations.
+
+Each resource have their own documentation, but this snippet gives a birds eye view how all the
+components integrate into a single picture when using IP Spaces:
+
+```hcl
+data "vcd_nsxt_manager" "main" {
+  name = "nsxManager1"
+}
+
+data "vcd_nsxt_tier0_router" "router" {
+  name            = "tier0Router-cloud"
+  nsxt_manager_id = data.vcd_nsxt_manager.main.id
+}
+
+data "vcd_org" "org1" {
+  name = "cloud"
+}
+
+data "vcd_org_vdc" "vdc1" {
+  org  = "cloud"
+  name = "nsxt-vdc-cloud"
+}
+
+resource "vcd_ip_space" "space1" {
+  name = "public-ip-space"
+  type = "PUBLIC"
+
+  internal_scope = ["192.168.1.0/24", "10.10.10.0/24", "11.11.11.0/24"]
+  external_scope = "0.0.0.0/24"
+
+  route_advertisement_enabled = false
+
+  ip_prefix {
```

Comment:
I have restructured it - you could see if it got any better

Reply:
I like it now, maybe one last request, would it be too verbose to say what exactly did the IP space do/use? The introduction chapter is really nice and explains it really well, but some more comments in the snippet, like `# This allocates a single IP address from the IP space`, also IMO it would represent the real world scenario better.

Reply:
Added more details

Reply:
Thanks!